source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,378 @@
+        {
+
+          prm.declare_entry ("Exclude composition", "-1",
+                             Patterns::Double (),
+                             "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                            );
+          prm.declare_entry ("List of isotherms", "",
+                             Patterns::List(Patterns::List(Patterns::Anything(),0,100000000,","),0,100000000,";"),
+                             "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                            );
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+        minimum_refinement_level = Utilities::string_to_int(prm.get ("Minimum refinement level"));
+        maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isotherms");
+        {
+          exclude_composition = Utilities::string_to_int(prm.get("Exclude composition"));
+          std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+          // process the List of isotherms to get all data out of the structure
+          isotherms.resize(isotherms_outer_loop.size(),std::vector<std::string>(4,"0"));
+          // loop through all given isotherms
+          for ( unsigned int isotherm_line = 0; isotherm_line != isotherms.size(); ++isotherm_line )
+            {
+              std::vector<std::string> isotherms_inner_loop = Utilities::split_string_list(isotherms_outer_loop[isotherm_line]);
+              // The minimum and maximum refinemnt levels respecively
+              isotherms[isotherm_line][0]=isotherms_inner_loop[0];
+              isotherms[isotherm_line][1]=isotherms_inner_loop[1];
+              // The temperatures for the min and max isotherms respectively
+              isotherms[isotherm_line][2]=isotherms_inner_loop[2];
+              isotherms[isotherm_line][3]=isotherms_inner_loop[3];
+
+            }
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    ASPECT_REGISTER_MESH_REFINEMENT_CRITERION(Isotherms,
+                                              "isotherms",
+                                              "A mesh refinement criterion that ensures a "
+                                              "maximum and minimum refinement level between "
+                                              "two temperatures (isotherms), with the possibility "
+                                              "to exclude a composition from this criterion. "
```

Comment:
I don't think I understand what "excluding a composition from this criterion" means. Do you want to say something like "with the possibility of excluding all cells from this criterion for which a particular compositional field has particular values"?

Reply:
I still think that it would be quite useful to expand on what exactly the exclusion does.